src/irr_v3d.h

```diff
@@ -29,5 +29,24 @@ typedef core::vector3d<s16> v3s16;
 typedef core::vector3d<u16> v3u16;
 typedef core::vector3d<s32> v3s32;
 
+#if __cplusplus >= 201103L
+
+#include <functional>
+
+// A hash class is required for c++11 unordered containers
+namespace std {
+	template<>
+	struct hash<v3s16> {
+		unsigned operator()(const v3s16 &pos) const
+		{
+			return (0x87654321 * pos.X)
+				^ (0x87654321 * pos.Y)
+				^ (0x87654321 * pos.Z);
```

Comment:
I tested it on sets of coordinates, and I have improved it by choosing different constants. With the new constants, collision-wise it performs comparable to the `std::string` hash function, and both are comparable to using random numbers as hash value. So, IMO now it should hardly be possible to find a better algorithm.

Note that the `std::string` hash function is much slower, probably because it requires constructing a `std::string`. 

For testing, I generated sets of coordinates, which I hashed with the different algorithms, and inserted into a hash table. I then counted the number of collisions (or actually: buckets with 0, 1, 2... entries) for each algorithm.

Sample output (columns: number of entries in bucket, number of buckets):

**Original algorithm**
```
Entries:  10125, buckets:  30000, zero:  22616, range:      0..    14, avg:      0.3, dev:    0.7
   0  [22616]  ==================================================
   1  [ 5337]  ===========
   2  [ 1599]  ===
   3  [  349]  
   4  [   67]  
   5  [   15]  
   6  [    2]  
   7  [    0]  
   8  [    0]  
   9  [    0]  
  10  [    0]  
  11  [    0]  
  12  [    9]  
  13  [    4]  
  14  [    2]  
```
**Revised algorithm:**
```
Entries:  10125, buckets:  30000, zero:  21396, range:      0..     5, avg:      0.3, dev:    0.6
   0  [21396]  ==================================================
   1  [ 7252]  ================
   2  [ 1199]  ==
   3  [  138]  
   4  [   14]  
   5  [    1]  
```
**`std::string` algorithm**
```
Entries:  10125, buckets:  30000, zero:  21420, range:      0..     4, avg:      0.3, dev:    0.6
   0  [21420]  ==================================================
   1  [ 7197]  ================
   2  [ 1235]  ==
   3  [  134]  
   4  [   14]  
```
**random values**
```
Entries:  10125, buckets:  30000, zero:  21394, range:      0..     4, avg:      0.3, dev:    0.6
   0  [21394]  ==================================================
   1  [ 7250]  ================
   2  [ 1209]  ==
   3  [  131]  
   4  [   16]  
```